.travis.yml

```diff
@@ -1,25 +1,60 @@
+image: php:7.1.1
 dist: trusty
-sudo: false
+sudo: true
 notifications:
   email: false
-language: node_js
-node_js:
-  - "8"
-  - "6"
-  - "4"
-  - "0.12"
-cache:
-  directories:
-    - node_modules
+language: php
+php:
+  # todo: add test support for other php versions
+  # - '5.6.31'
+  # - '7.0.22'
+  - '7.1.12'
+env:
+  NODE_VERSION=8
+  NODE_VERSION=6
+  NODE_VERSION=4
+  NODE_VERSION=0.12
 before_script:
-  - "npm run build"
-script:
-  - "npm test"
+  - sudo add-apt-repository -y ppa:ondrej/php
+  - sudo apt-get update -y
+  - sudo apt-get install -y apache2 php7.1 libapache2-mod-php7.1 php7.1-cli php7.1-common php7.1-mbstring php7.1-gd php7.1-intl php7.1-xml php7.1-mysql php7.1-mcrypt php7.1-zip
 jobs:
   include:
+    - stage: npm 8
+      script:
+        - nvm install 8
+        - nvm use 8
+        - npm install
+        - npm run build
+        - npm test
+    - stage: npm 6
+      script:
+        - nvm install 6
+        - nvm use 6
+        - npm install
+        - npm run build
+        - npm test
+    - stage: npm 4
+      script:
+        - nvm install 4
+        - nvm use 4
+        - npm install
+        - npm run build
+        - npm test
+    - stage: npm 0.12
+      script:
+        - nvm install 0.12
+        - nvm use 0.12
+        - npm install
+        - npm run build
+        - npm test
     - stage: lint and npm release
-      node_js: "8"
       script:
+        - nvm install 8
+        - nvm use 8
+        - npm install
+        - npm run build
+        - npm test
         - npm run lint
       after_success:
         - TRAVIS_JOB_NUMBER=WORKAROUND.1 npm run semantic-release
```

Comment:
I assume that all build stages run sequentially and not in parallel? If so, put nodejs@8 last and have it run the `after_success` target to publish - no need to do this twice.